### PR TITLE
log unchecked error

### PR DESCRIPTION
**Pull request by bazile-clyde**
# Description
This is the only place in this function where we don't check the error message and just silently keep trying to grab media. Maybe adding this to the debug logs will give us some insight as to why streams go down.

# Testing
I tested this by replacing `mediaStream.Next` with my own function that returns an error for short period of time.

```
// var t = time.Now() // global var
...
			media, release, err := func() (T, func(), error) {
				since := time.Since(t).Seconds()
				if since > 10*time.Second.Seconds() && since < 11*time.Second.Seconds() {
					return *new(T), func() {}, errors.New("MY ERROR")
				}
				return mediaStream.Next(ctx)
			}()

```
That should allow you to see something like 
```
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
x264 [info]: profile Constrained Baseline, level 3.0, 4:2:0, 8-bit
2024-01-10T17:31:31.771-0500    DEBUG   gostream/source_stream_utils.go:97      unchecked error {"error": "MY ERROR"}
2024-01-10T17:31:31.772-0500    DEBUG   gostream/source_stream_utils.go:97      unchecked error {"error": "MY ERROR"}
2024-01-10T17:31:31.772-0500    DEBUG   gostream/source_stream_utils.go:97      unchecked error {"error": "MY ERROR"}
... continue logging the error for one second ...
2024-01-10T17:31:51.233-0500    INFO    gostream/stream.go:268  detected new image bounds       {"width": 640, "height": 480}
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
x264 [info]: profile Constrained Baseline, level 3.0, 4:2:0, 8-bit
^C2024-01-10T17:32:01.684-0500  INFO    videosource     logging/logger.go:142   Terminating videosource logger
```

### CODE REVIEW

File: gostream/cmd/stream_av/main.go [hunk 1]
```diff
@@ -128,8 +130,9 @@ func runServer(
 		err = multierr.Combine(err, <-audioErr, server.Stop(ctx))
 	}()
 
+	log := logging.NewLogger("stream_av")
```
Comment (bazile-clyde) [inline gostream/cmd/stream_av/main.go:133]: We don't use these files very much, so I just added in a quick fix for the new logger situation.

File: gostream/source_stream_utils.go [hunk 2]
```diff
@@ -83,6 +86,7 @@ func streamMediaSource[T, U any](
 			}
 			media, release, err := mediaStream.Next(ctx)
 			if err != nil {
```
Comment (bazile-clyde) [inline gostream/source_stream_utils.go:88]: adding
```
error = errors.New("TESTING")
```
before this line is how I tested that below only show up when debug mode is on. Weirdly enough, it also breaks _just_ the live stream.

**Review by seanavery — APPROVED** [review]
Awesome to get more visibility here. Makes sense to me to add logging visibility in `streamMediaSource`. Currently `streamLoop` just continues silently when `Next` call fails.

**Comment by bazile-clyde** [comment]
Closing since this is _not_ the cause or would've helped with https://viam.atlassian.net/browse/RSDK-6287

**Pull request closed without merging**